### DISCUSSION TITLE
Prefab에 붙은 ZepetoScript의 변수 초깃값이 적용되지 않습니다.

### DISCUSSION THREAD

**Discussion by Cha-SeongHun** [naverz/zepeto-studio-kor#662] [Issue Report (Legacy)]
씬에 배치된 프리팹의 스크립트에서 아래와 같이 TimeOffset 을 0.5로 변경합니다.
프리팹 내에서 TimeOffset은 0으로 설정되어 있습니다.
![image](https://user-images.githubusercontent.com/100827088/165225337-2a8815af-51f9-42d6-95fb-bbc4979599f5.png)

이후 Play를 누르면 TimeOffset이 프리팹 기본값을 따라 0으로 자동 변경되는 모습을 볼 수 있습니다.
![image](https://user-images.githubusercontent.com/100827088/165225103-17cc678e-d8ae-4cb1-b4a7-a00fa44d59bf.png)

이 증상으로 인해 스크립트의 변수를 에디터에서 조작하는 것이 의미가 없는 상황입니다.
프리팹을 사용하지 않는 방법으로 회피 가능하나 레벨디자인이 매우 어려워져 적합한 대처방법은 아니라고 생각되네요.

제페토 패키지 버전은 1.2.0이며, 1.1.5 때부터 쭉 존재해 온 이슈입니다.
확인해주시면 감사하겠습니다

**Comment by x4cf** [comment]
코드를 한번 볼 수 있을까요?
선언할때 초기화해주고 인스팩터에서 변경하면 적용되기도 해요 :)

**Reply by Cha-SeongHun** [reply]
아래처럼 코드에 변수만 존재하고 아무것도 없어도 재현됩니다.

```ts
export default class TestScript extends ZepetoScriptBehaviour {
  public foo: float;
}
```

**Reply by x4cf** [reply]
export default class TestScript extends ZepetoScriptBehaviour {
  public foo: float = 0;
}

이렇게 하면 해결되기도 해요 :)

**Reply by Cha-SeongHun** [reply]
음.. 초기화 유무에 상관없이 발생합니다

**Comment by Andwhy81kr** [comment]
https://cafe.naver.com/zepetostudio/15
https://youtu.be/7ARUGVwXMGs

이문제죠??

해당문제에 대해 버그리포팅을 하긴했는데, 수정예정이라고는 했지만, 아직 안되었나보네요.
그냥 버그를 회피하는방법이 가장 빠릅니다.

**Reply by Cha-SeongHun** [reply]
회피하려면 프리팹을 사용하지 않는 수밖에 없는데, 다른 방법이 있나요??

**Reply by Andwhy81kr** [reply]
이야기 하신것처럼 프리팹을 배치해놓고 다시 프리팹을 뽀개 버리던지,
아니면 이 프리팹의 인스팩터 값을 설정해주는 다른 컨트롤러 객체를 하나 더 두어서 만들어서 사용하고 있습니다.
